### PR TITLE
Add bcrypt module definitions

### PR DESCRIPTION
**Pull request by yamadayuki**
Add types for `bcrypt` module.
refs. https://github.com/ncb000gt/node.bcrypt.js#api

### CODE REVIEW

File: definitions/npm/bcrypt_v0.8.x/flow_>=v0.29.x/bcrypt_v0.8.x.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+declare module bcrypt {
+  declare function genSaltSync(rounds?: number): string;
+  declare function genSalt(rounds: number, callback: (err: Error, salt: string) => void): void;
+  declare function genSalt(callback: (err: Error, salt:string) => void): void;
+  declare function hashSync(data: any, salt: string): string;
+  declare function hashSync(data: any, rounds: number): string;
+  declare function hash(data: any, salt: string, callback: (err: Error, encrypted: string) => void): void;
```
Comment (thejameskyle) [inline definitions/npm/bcrypt_v0.8.x/flow_>=v0.29.x/bcrypt_v0.8.x.js]: It seems like `data` is supposed to just be a string. Is that not the case?

Comment (yamadayuki) [inline definitions/npm/bcrypt_v0.8.x/flow_>=v0.29.x/bcrypt_v0.8.x.js]: Thank you for your advice!
I confirmed what the data type is. The `data` must be a string.
```js
const salt = bcrypt.genSaltSync();

bcrypt.hash('yamadayuki', salt, (err, encrypted) => {
  if (err) throw err;     // no error
  console.log(encrypted); // outputs encrypted string
});

bcrypt.hash(197, salt, (err, encrypted) => {
  if (err) throw err;     // throw error
  console.log(encrypted); // nope
});

> Error: data must be a string and salt must either be a salt string or a number of rounds

```

I'll fix it 🚀

**Comment by thejameskyle** [comment]
👍

**Comment by yamadayuki** [comment]
Thanks!

**Pull request merged**